File: src/main/java/com/mexchange/api/entity/ExchangePool.java
package com.mexchange.api.entity;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.math.BigDecimal;

@NoArgsConstructor
@Getter
@Setter
@Document(collection = "exchangePools")
public class ExchangePool {
    @Id
    private String id;

    private String name;

    private CurrencyPool firstCurrency;
    private CurrencyPool secondCurrency;

    private Rate rate;

    @NoArgsConstructor
    @Getter
    @Setter
    public static class CurrencyPool {
        private String currency;
        private BigDecimal availableAmount;
    }

    @NoArgsConstructor
    @Getter
    @Setter
    public static class Rate {
        private String currency;
        private BigDecimal amount;
    }
}

File: src/main/kotlin/com/mexchange/api/entity/ExchangePool.kt
package com.mexchange.api.entity

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.math.BigDecimal


@Document(collection = "exchangePools")
data class ExchangePool(
    @Id var id: String? = null,
    var name: String? = null,
    var firstCurrency: CurrencyPool? = null,
    var secondCurrency: CurrencyPool? = null,
    var rate: Rate? = null
) {

    data class CurrencyPool (
        var currency: String? = null,
        var availableAmount: BigDecimal? = null
    )

    data class Rate (
        var currency: String? = null,
        var amount: BigDecimal? = null
    )
}
